File: Web/src/services/notificationService.js
import { apiRequest } from './apiClient';

export async function getNotifications() {
  const response = await apiRequest('/notifications');
  return {
    unreadCount: response?.data?.unreadCount ?? 0,
    notifications: response?.data?.notifications || [],
  };
}

export async function markNotificationRead(id) {
  const response = await apiRequest(`/notifications/${id}/read`, {
    method: 'PATCH',
  });
  return response?.data?.notification;
}

export async function markAllNotificationsRead() {
  return apiRequest('/notifications/read-all', {
    method: 'PATCH',
  });
}

File: App/lib/services/notification_service.dart
import '../core/constants/api_endpoints.dart';
import '../core/network/api_client.dart';
import '../models/notification_model.dart';

class NotificationListResult {
  NotificationListResult({
    required this.notifications,
    required this.unreadCount,
  });

  final List<AppNotification> notifications;
  final int unreadCount;
}

class NotificationService {
  NotificationService({required ApiClient apiClient}) : _api = apiClient;

  final ApiClient _api;

  Future<NotificationListResult> getNotifications() async {
    final response = await _api.get(ApiEndpoints.notifications);
    final data = response['data'] as Map<String, dynamic>? ?? {};
    final list = (data['notifications'] as List? ?? [])
        .whereType<Map>()
        .map((e) => AppNotification.fromJson(Map<String, dynamic>.from(e)))
        .toList();

    return NotificationListResult(
      notifications: list,
      unreadCount: (data['unreadCount'] as num?)?.toInt() ?? 0,
    );
  }

  Future<int> getUnreadCount() async {
    final response = await _api.get(ApiEndpoints.unreadNotifications);
    final data = response['data'] as Map<String, dynamic>? ?? {};
    return (data['unreadCount'] as num?)?.toInt() ?? 0;
  }

  Future<AppNotification> markRead(String id) async {
    final response =
        await _api.patch('${ApiEndpoints.notifications}/$id/read');
    final data = response['data'] as Map<String, dynamic>? ?? {};
    return AppNotification.fromJson(
      Map<String, dynamic>.from(data['notification'] as Map),
    );
  }

  Future<void> markAllRead() async {
    await _api.patch(ApiEndpoints.readAllNotifications);
  }
}
